common/ts/__tests__/publish-task-test.ts

```diff
@@ -0,0 +1,119 @@
+import Analysis from '../sonarqube/Analysis';
+import Endpoint, { EndpointType } from '../sonarqube/Endpoint';
+import Metrics from '../sonarqube/Metrics';
+import Task, { TimeOutReachedError } from '../sonarqube/Task';
+import TaskReport from '../sonarqube/TaskReport';
+import * as publishTask from '../publish-task';
+import * as serverUtils from '../helpers/vsts-server-utils';
+import * as tl from 'vsts-task-lib/task';
+import { InvalidApiResourceVersionError } from 'vso-node-api/VsoClient';
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+});
+
+const TASK_REPORT = new TaskReport({
+  ceTaskId: '111',
+  ceTaskUrl: 'http://taskurl1',
+  dashboardUrl: 'http://dashboardurl1',
+  projectKey: 'projectKey1',
+  serverUrl: 'http:/serverUrl1'
+});
+
+const SC_ENDPOINT = new Endpoint(EndpointType.SonarCloud, { url: 'https://endpoint.url' });
+const SQ_ENDPOINT = new Endpoint(EndpointType.SonarQube, { url: 'https://endpoint.url' });
+const METRICS = new Metrics([]);
+
+it('should fail unless SONARQUBE_SCANNER_PARAMS are supplied', async () => {
+  jest.spyOn(tl, 'getVariable').mockImplementation(() => null);
+  jest.spyOn(tl, 'setResult')
+
+  await publishTask.default(EndpointType.SonarCloud);
+
+  expect(tl.getVariable).toBeCalledWith('SONARQUBE_SCANNER_PARAMS');
+  expect(tl.setResult).toBeCalledWith(tl.TaskResult.Failed,
+    'The SonarCloud Prepare Analysis Configuration must be added.');
+});
+
+it('get report string for single report', async () => {
+  // Mock waiting for the ceTask to complete and return a Task
+  const returnedTask = new Task({ analysisId: '123', componentKey: 'key', status: 'status', type: EndpointType.SonarQube, componentName: 'componentName' });
+  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(() => returnedTask);
+
+  // Mock converting the Task into an html report
+  const returnedAnalysis = new Analysis({ status: '', conditions: [] }, EndpointType.SonarCloud, '', null, null);
+  jest.spyOn(Analysis, 'getAnalysis').mockImplementation(() => returnedAnalysis);
+  jest.spyOn(returnedAnalysis, 'getHtmlAnalysisReport').mockImplementation(() => 'dummy html');
+
+  const result = await publishTask.getReportForTask(TASK_REPORT, METRICS, SQ_ENDPOINT, 999);
+
+  expect(Task.waitForTaskCompletion).toHaveBeenCalledWith(SQ_ENDPOINT, TASK_REPORT.ceTaskId, 999);
+  expect(Analysis.getAnalysis).toBeCalledWith(
+    { analysisId: '123', dashboardUrl: 'http://dashboardurl1', endpoint: SQ_ENDPOINT, metrics: { metrics: [] }, projectName: 'componentName' });
```

Comment:
This test checks the new behaviour i.e. that the component/project name is always supplied when creating the Analysis object.